add auth rfc (#8)
elorzafe/amplify-types: RFC.md

# `Auth` Changes

> Amplify is proposing the following changes for the `Auth` category.

Reviewer: ```suggestion
Amplify is proposing the following changes for the `Auth` category. Similar changes will be applied across all of the `Auth` APIs but examples for specific APIs are highlighted below.
```

Added some wording to make it clear these changes will be applied across auth (which customers can experiment with in the playground)

> ## Typescript support for user attributes

Reviewer: ```suggestion
## TypeScript support for user attributes
```

User attributes inference on the `signUp` API.

**Amplify v5 (`aws-amplify@5`)**

> ```Typescript
> 
>   Auth.signUp({
> 	  username:"username",
>       password:"*******",
> 	  attributes:{
> 		email: "email@email"
> 	 }})

Reviewer: ```suggestion
```Typescript
Auth.signUp({
  username: "username",
  password: "*******",
  attributes: {
    email: "email@domain.com"
  }
});
```

```

**Amplify v5 DX**

Todo: add screen shoots

**Proposed Amplify v6 (`aws-amplify@6`)**

```Typescript

> 
>   signUp({
>     username: "username",
>     password: "********",
>     options: {
>       userAttributes: [{ userAttributeKey: "email", value: "email@email.com" }],
>     },
>   })
> 

Reviewer: ```suggestion
signUp({
  username: "username",
  password: "********",
  options: {
    userAttributes: [{ userAttributeKey: "email", value: "email@domain.com" }],
  },
});
```

```

**Proposed Amplify v6 DX**

Todo: add screen shoots

## Predictable API responses

> We are improving **_DX_** by providing descriptive API responses that help customers to complete auth flows.

Reviewer: ```suggestion
We are improving **_DX_** by providing descriptive API responses to help customers complete auth flows. An example for the `confirmSignUp` API is highlighted below.
```

**Amplify v5 (`aws-amplify@5`)**

```Typescript

> 
> const resp = Auth.confirmSignUp("username", "112233")

Reviewer: ```suggestion
const resp = Auth.confirmSignUp("username", "112233");
```

```

**Amplify v5 DX**

Todo: add screen shoots

**Proposed Amplify v6 (`aws-amplify@6`)**

```Typescript

> 
>   const resp = await confirmSignUp({
>     username: "username",
>     confirmationCode: "112233",
>   });
> 
>   if (resp.isSignUpComplete) {
>     // Show login component
>   }
> 

Reviewer: ```suggestion
const resp = await confirmSignUp({
  username: "username",
  confirmationCode: "112233",
});
```

Removed the `resp.isSignUpComplete` part from my suggestion, but I'm okay with it as long as we also show the equivalent flow for v5 (consistency between samples).

Author: just added a similar flow to v5
